bpf/lib/nodeport_egress.h

```diff
@@ -393,6 +403,8 @@ static __always_inline int nodeport_snat_fwd_ipv4(struct __ctx_buff *ctx,
 __section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_IPV4_NODEPORT_SNAT_FWD)
 int tail_handle_snat_fwd_ipv4(struct __ctx_buff *ctx)
 {
+	__u32 __maybe_unused src_sec_identity = UNKNOWN_ID;
+	__u32 magic = ctx->mark & MARK_MAGIC_HOST_MASK;
 	__u32 cluster_id = ctx_load_and_clear_meta(ctx, CB_CLUSTER_ID_EGRESS);
```

Comment:
I feel we'd better use the same way as `cluster_id = ctx_load_and_clear_meta(ctx, CB_CLUSTER_ID_EGRESS);`:

```diff
- __u32 magic = ctx->mark & MARK_MAGIC_HOST_MASK;
+ __u32 src_sec_identity = ctx_load_and_clear_meta(ctx, CB_SRC_LABEL);
```

Also, this requires CB_SRC_LABEL to be set from callers, e.g. `handle_nat_fwd` https://github.com/cilium/cilium/blob/v1.17.0/bpf/lib/nodeport_egress.h#L712
```diff
+ ctx_store_meta(ctx, CB_SRC_LABEL, src_id);
```
the `src_id` is already known in the caller of `handle_nat_fwd`, just pass it in and store it in cb.

Reply:
Thanks for this observation. I just updated the code following this style and made sure that the outcome was the same. Let me know :smiley: